server/dns-service/dnsservice-dnsjava/src/main/java/org/apache/james/dnsservice/dnsjava/DNSJavaService.java

```diff
@@ -141,16 +142,16 @@ public void configure(HierarchicalConfiguration<ImmutableNode> configuration) th
         List<Name> sPaths = new ArrayList<>();
         if (autodiscover) {
             LOGGER.info("Autodiscovery is enabled - trying to discover your system's DNS Servers");
-            String[] serversArray = ResolverConfig.getCurrentConfig().servers();
+            List<InetSocketAddress> serversArray = ResolverConfig.getCurrentConfig().servers();
             if (serversArray != null) {
-                for (String aServersArray : serversArray) {
-                    dnsServers.add(aServersArray);
+                for (InetSocketAddress aServersArray : serversArray) {
+                    dnsServers.add(aServersArray.getHostName());
```

Comment:
are you sure you want the hostname here?

Reply:
in doubt I would have used `getHostString`

Reply:
`                    dnsServers.add(aServersArray.getAddress().toString());`

Gets the tests failing:

```
[ERROR] o.a.j.d.d.DNSJavaService - DNS service could not be initialized.  The DNS servers specified are not recognized hosts.
java.net.UnknownHostException: /127.0.0.53: Name or service not known
	at java.base/java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
```

I imagine my above snipset attemps a resolution.

`dnsServers.add(aServersArray.getHostString());` also works.

What do you prefer?

Reply:
(please note that nothing forbids to pass an IP address as an hostname)

Reply:
if you pass an IP address getHostName will attempt a resolution. Do we want that?

Reply:
```

String | getHostString() Returns the hostname, or the String form of the address if it  doesn't have a hostname (it was created using a literal).
```

Let's go for that option.